spec/controllers/authorities_controller_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe AuthoritiesController, :type => :controller do
+  describe '#query', :vcr do
+    context 'invalid query' do
+      subject { get :query }
+      
+      it { is_expected.to have_http_status(:success) }
+      
+      it 'returns empty array' do
+        expect(subject["response"]).to be_nil
+      end
+    end
+    
+    context 'lcsh' do
+      subject { JSON.parse(get(:query, q: 'coffee', term: 'lcsh').body) }
+      
+      it 'returns values from LOC' do
+        expect(subject.count).to eq(10)
+        
+        subject.each{ |s|
+          expect(s.downcase).to include('coffee')
+        }
+      end
+    end
+    
+    context 'subject_geographic' do
+      subject { JSON.parse(get(:query, q: 'ethiopia', term: 'subject_geographic').body) }
+      
+      it 'returns values from FAST' do
+        expect(subject.count).to eq(10)
+        
+        subject.each{ |s|
+          expect(s.downcase).to include('ethiopia')
+        }
+      end
+    end
+    
+    context 'based_near' do
+      subject { JSON.parse(get(:query, q: 'ethiopia', term: 'based_near').body) }
+      
+      it 'returns values from GeoNames' do
+        expect(subject.count).to eq(10)
+        subject.each{ |s|
+          expect(s["value"].downcase).to include('ethiopia')
+        }
+      end
+    end
+  end
```

Comment:
Missing tests for queries not returning results and empty queries.

Reply:
we require at least two characters before making the request, should i still test for empty queries?

Reply:
Nah